sdks/android/Mentat/library/src/main/java/org/mozilla/mentat/RustObject.java

```diff
@@ -35,7 +42,25 @@ void validate() {
         }
     }
 
-    public Pointer getPointerForUUID(UUID uuid) {
+    Pointer validPointer() {
+        this.validate();
+        return this.rawPointer;
+    }
+
+    boolean isConsumed() {
```

Comment:
nit: Perhaps use `isConsumed` in `validate`, since concept of "valid" is defined in terms of "consumed"?

Reply:
And perhaps rename `validate` into something like `assertValidPointer`. Would also be good to establish a pattern (in a docstring) as to when we should throw on null pointer, and when we simply check.